### PR TITLE
Fix CardView related crash in Giphy ViewHolder

### PR DESCRIPTION
**Pull request by ogkuzmin**
[CAS-525](https://stream-io.atlassian.net/browse/CAS-525?atlOrigin=eyJpIjoiMWE2YTJiZjkzNDk1NDBmMWJhNDVjZjA4OTExMTEwZGYiLCJwIjoiaiJ9)

### Description

Fix crash with MaterialCardView for old android versions, by using `app:cardBackgroundColor` instead of `android:background`. Also fix the colour being used for the background.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] <del> Changelog updated with client-facing changes </del>
- [ ] <del> New code is covered by unit tests </del>
- [ ] <del> Comparison screenshots added for visual changes </del>
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt [hunk 1]
```diff
@@ -68,18 +69,22 @@ internal class BackgroundDecorator : BaseDecorator() {
...
 
     override fun decorateGiphyMessage(viewHolder: GiphyViewHolder, data: MessageListItem.MessageItem) {
-        viewHolder.binding.cardView.shapeAppearanceModel = ShapeAppearanceModel.builder()
+        viewHolder.binding.cardView.background = ShapeAppearanceModel.builder()
             .setAllCornerSizes(DEFAULT_CORNER_RADIUS)
             .setBottomRightCornerSize(SMALL_CARD_VIEW_CORNER_RADIUS)
             .build()
+            .let(::MaterialShapeDrawable)
+            .apply {
+                setTint(ContextCompat.getColor(viewHolder.context, R.color.stream_ui_background_default))
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt:86]: We gave a `getColorCompat` extension for this

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_giphy.xml [hunk 2]
```diff
@@ -14,12 +14,11 @@
         app:layout_constraintTop_toTopOf="parent"
         />
 
-    <com.google.android.material.card.MaterialCardView
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_giphy.xml:17]: What was causing the crash exactly? `MaterialCardView` is a library class, it should work on all API versions, shouldn't it?

And if we do really have to remove it, we should also check the other two places in the project where we're using it.

Comment (ogkuzmin) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_giphy.xml:17]: It doesn't work with the background option somewhy. Exception happened in LayoutInflater class. MaterialCardView works on all API versions, except in combination with that attribute. I couldn't find any similar issue on Github with this problem

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt (hunk 1 above)
Comment (ogkuzmin) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt]: Changed

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_giphy.xml (hunk 2 above)
Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_giphy.xml:17]: Gonna look into this a bit

**Review by zsmb13 — APPROVED** [review]

**Comment by ogkuzmin** [comment]
@zsmb13 good job!

**Pull request merged**